File: GFG/Arrays/maxdifference.py
# find the maximum differnece arr[j]-arr[i] such that j>i;

def maxdif(arr):
    j=1
    res = arr[1]-arr[0]
    minv=arr[0]
    while j<len(arr):
        res = max(res,(arr[j]-minv))
        minv=min(minv,arr[j])
        j+=1
    return res

brr=[7,9,5,6,3,2]
print(maxdif(brr))
        
        
        

File: GFG/Arrays/maxdifference.cpp
#include<iostream>
#include<vector>
using namespace std;

int maxdiff(vector<int> &arr){
    int res=arr[1]-arr[0];
    int minv=arr[0];
    for(int i = 1;i<arr.size();i++){
        res=max(res,arr[i]-minv);
        minv = min(minv,arr[i]);
    }
    return res;
}

int main(){
    vector<int> arr = {7,9,5,6,3,2};
    cout<<maxdiff(arr);
    return 0;
}